### DISCUSSION TITLE
Implementing a `zbm.await=` option?

### DISCUSSION THREAD

**Discussion by jrtoddy** [zbm-dev/zfsbootmenu#271] [Feature Requests]
Would it be acceptable to implement a KCL option that functions like `zbm.prefer=!` but simply waits for the pool to be visible? I'd be happy to have a go at implementing it myself...

Reason being, I'd like to be able to use whole disk encryption on my boot pool, with no EFI partitions and the like cluttering them up, and have a bootable USB stick with a "keyring" pool able to provide the decryption key but not interfere with the `zbm.prefer` behavior later in the chain.

So, the KCL could look something like this if required.

`ro loglevel=4 nomodeset zbm.await=keyring zbm.prefer=tank! ...`

Let me know what you think. :+1:

**Comment by zdykstra** [comment]
This isn't something that I'd like to see as part of the main ZBM codebase right now. However, it should be fully possible for you to implement a solution that can be added to `contrib/` . Early hooks are executed right before any pool import attempts take place:

https://github.com/zbm-dev/zfsbootmenu/blob/master/zfsbootmenu/libexec/zfsbootmenu-init#L78-L86

You'll want to source in a few helper libraries to access the same functions that `zfsbootmenu-init` can use.  I'm happy to help with a general framework for this, if you need.

**Reply by jrtoddy** [reply]
Thanks for the advice, I'll plan to make an optional `contrib/` element that can implement this feature once my bits of work with `zbm-builder.sh` reach a natural pause. I'm sure I'll need some help but let me have a crack at it as a learning exercise first. :wink:
